### PR TITLE
backport: bitcoin#15584, #16060, #16465, #16477, #16680, #16714, #16941, #16991, #17453, #17696, #19058 (last batch for v19)

### PR DESCRIPTION
**Pull request by knst**
## What was done?
Last batch of missing backports from v18-v19 and related changes.

 - bitcoin#16477
 - bitcoin#16680
 - bitcoin#16714
 - bitcoin#17696
 - bitcoin#17453
 - bitcoin#15584
 - bitcoin#19058
 - bitcoin#16941
 - bitcoin#16991
 - bitcoin#16465
 - bitcoin#16060

To follow-up backport bitcoin#16060 there were buried also 2 dash's specific deployments:
 - DIP0001
 - DIP0008

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
No breaking changes

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
lightly tested ACK

File: src/chainparams.cpp [hunk 1]
```diff
@@ -191,13 +191,6 @@ class CMainParams : public CChainParams {
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0003].nWindowSize = 4032;
         consensus.vDeployments[Consensus::DEPLOYMENT_DIP0003].nThresholdStart = 3226; // 80% of 4032
 
-        // Deployment of DIP0008
-        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0008].bit = 4;
-        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0008].nStartTime = 1557878400; // May 15th, 2019
-        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0008].nTimeout = 1589500800; // May 15th, 2020
-        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0008].nWindowSize = 4032;
-        consensus.vDeployments[Consensus::DEPLOYMENT_DIP0008].nThresholdStart = 3226; // 80% of 4032
```
Comment (PastaPastaPasta) [inline src/chainparams.cpp:210]: other deployments have been buried as well no? Should do those in another PR at least

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit; however a subsequent PR should be done to bury the other hardened hard forks
